tests/RL_loop_test.py

```diff
@@ -0,0 +1,99 @@
+from recommence.Checkpoint import Checkpoint
+from _utils.TileCodingAgent import TileCodingAgent
+from RlGlue import RlGlue, BaseEnvironment
+import gymnasium
+from typing import Any, Dict, Tuple
+
+class stepCollector:
+    def __init__(self):
+        self.total_steps = 0
+
+    def add(self, steps):
+        self.total_steps += steps
+
+    def get(self):
+        return self.total_steps
+
+    def reset(self):
+        self.total_steps = 0
+
+
+class rewardCollector:
+    def __init__(self):
+        self.total_reward = 0
+
+    def add(self, r):
+        self.total_reward += r
+
+    def get(self):
+        return self.total_reward
+
+    def reset(self):
+        self.total_reward = 0
+
+
+class CartpoleEnvironment(BaseEnvironment):
+    def __init__(self, seed: int):
+        self.env = gymnasium.make("CartPole-v1", max_episode_steps=1000000)
+        self.seed = seed
+
+    def start(self):
+        self.seed += 1
+        s, info = self.env.reset(seed=self.seed)
+        return s
+
+    def step(self, action) -> Tuple[float, Any, bool, Dict[str, Any]]:
+        sp, reward, terminal, truncate, info = self.env.step(action)
+        return float(reward), sp, terminal, {}
+
+
+def test_RL_integration(tmp_path):
```

Comment:
Here's how I recommend we do this test: let's run the RL system for 10k steps with no checkpointing at all and return (1) the agent's weights and (2) the sequence of returns observed.

Then we run the system a second time with checkpointing enabled (e.g. at 5k steps) and return the same two things and we check if the weights and returns are exactly identical between both runs.

So at a high-level, the code would look something like:
```python
class FakeCheckpoint:
  def register(self, key: str, builder: ...):
    return builder()

def run_rl_system(seed: int, should_chk: bool, tmp_path) -> Tuple[np.ndarray, ReturnCollector]:
  chk = Checkpoint(str(tmp_path)) if should_chk else FakeCheckpoint()

  agent = chk.register('agent', lambda: TileCodingAgent())
  ...

  while running:
    if steps == 5k and should_chk:
      del chk
      chk = Checkpoint(tmp_path)
      agent = chk['agent']
      ...

  return agent.weights, return_collector

def test_RL_integration(tmp_path):
  no_chk_result = run_rl_system(seed=42, should_chk=False, tmp_path=tmp_path)
  chk_result = run_rl_system(seed=42, should_chk=True, tmp_path=tmp_path)

  assert np.all(no_chk_result[0] == chk_result[0])
```

Reply:
That makes sense. 

We don't have that _save_every() function that saves everything periodically and behind the curtains, yet, so I just added `chk.save()` before deleting it for now.